### PR TITLE
#235 Add schema export button

### PR DESCRIPTION
**Pull request by GeorgiChochov**

### CODE REVIEW

**Comment by lokm01** [comment]
Awesome, this will be useful particularly for the cobol use cases ;)

**Comment by Zejnilovic** [comment]
## 1 Cobol copybook has a `.json` file extension
When exported Cobol copybook is `SchemanName-Version.json` and then there are clashes with predefined tools for JSONs and need to rename.

**Comment by GeorgiChochov** [comment]
It might make sense for me to upload a file and download it with a familiar name, but if somebody else downloads it that name might not make sense (or be particularly appropriate). I'll name the file based on the schema name and version, with an extension based on the MIME type.

File: menas/src/main/scala/za/co/absa/enceladus/menas/controllers/SchemaController.scala [hunk 1]
```diff
@@ -72,6 +74,19 @@ class SchemaController @Autowired() (
...
 
+  @GetMapping(path = Array("/export/{name}/{version}"))
+  @ResponseStatus(HttpStatus.OK)
+  def exportSchema(@AuthenticationPrincipal principal: UserDetails,
+                   @PathVariable name: String,
+                   @PathVariable version: Int,
+                   response: HttpServletResponse): CompletableFuture[Array[Byte]] = {
+    attachmentService.getSchemaByNameAndVersion(name, version).map { attachment =>
+      response.addHeader("filename", attachment.filename)
+      response.setContentType(attachment.fileMIMEType)
```
Comment (GeorgiChochov) [inline menas/src/main/scala/za/co/absa/enceladus/menas/controllers/SchemaController.scala]: This gets overridden by the fact that we are returning `Array[Byte]` and is always set to `application/octet-stream`.
I should write tests for this API endpoint.

File: menas/ui/components/schema/schemaDetail.controller.js [hunk 2]
```diff
@@ -109,6 +110,17 @@ sap.ui.define([
         })
     },
 
+    _getFileExtension: function (mimeType) {
```
Comment (Zejnilovic) [inline menas/ui/components/schema/schemaDetail.controller.js]: Seems like the cob isn't added there
image

I am uploading a copybook that is `NAMEXX.txt` and mongo says `{ "fileMIMEType": "text/plain" }`

Comment (Zejnilovic) [reply menas/ui/components/schema/schemaDetail.controller.js]: Actually, even JSON now downloads a txt

Comment (GeorgiChochov) [reply menas/ui/components/schema/schemaDetail.controller.js]: Mine have `.json` going in and `.json` coming out. Everything that has `.txt` comes out `.txt`. Things that have `.cob` or no extension going in, come out `.cob`. I'm not in control of the MIME type coming in...
What is your suggestion?

Comment (Zejnilovic) [reply menas/ui/components/schema/schemaDetail.controller.js]: Not a suggestion but an issue 
Everything downloads as plain text file

File: menas/src/main/scala/za/co/absa/enceladus/menas/repositories/AttachmentMongoRepository.scala [hunk 3]
```diff
@@ -15,17 +15,48 @@
...
+  }
+
+  private def getByCollectionAndNameAndVersion(refCollection: String,
+                                               name: String,
+                                               version: Int): Future[Option[MenasAttachment]] = {
+
+    val pipeline = Seq(
+      filter(equal("refCollection", refCollection)),
+      filter(equal("refName", name)),
+      project(Document(s"{diff: {$$subtract: [$version, '$$refVersion']}, doc: '$$$$ROOT'}")),
```
Comment (yruslan) [inline menas/src/main/scala/za/co/absa/enceladus/menas/repositories/AttachmentMongoRepository.scala]: 👍 Nice query

File: menas/ui/components/schema/schemaDetail.controller.js (hunk 2 above)
Comment (GeorgiChochov) [reply menas/ui/components/schema/schemaDetail.controller.js]: It happens, thanks for testing!

**Pull request merged**